resilience4j-retry/src/main/java/io/github/resilience4j/retry/RetryConfig.java

```diff
@@ -90,6 +95,18 @@ public int getMaxAttempts() {
         return intervalFunction;
```

Comment:
I think we should deprecate `getIntervalFunction`, since it shouldn't be used anymore in our  `RetryImpl`. Makes it more obvious where it is still used. 
Comparable to `getWaitDurationInOpenState` in `CircuitBreakerConfig`

Reply:
This was one of my questions from PR description 😃

Reply:
I misread the PR description. I thought you want to deprecate `IntervalFunction` itself. 
Too many topics in parallel, sorry.

Reply:
no worries